src/lib/containers/DirectDownload.tsx

```diff
@@ -30,35 +43,73 @@ const DirectDownload: React.FunctionComponent<DirectFileDownloadProps> = (props)
     }
   }, [token])
 
-  const getDownloadLink = () => {
-    SynapseClient.getFileResult(
-      fileEntity!,
-      token,
-      false,
-      true
-    ).then(data => {
-      if (data.preSignedURL) {
-        window.open(data.preSignedURL)
+  const getDownloadLink = async () => {
+    let preSignedURL
+    try {
+      if (associatedObjectType === FileHandleAssociateType.TableEntity) {
+        const files = await getTableEntityFileHandle()
+        preSignedURL = files.requestedFiles[0].preSignedURL!
+      } else {
+        const file = await getFileResult(
+          fileEntity!,
+          token,
+          false,
+          true
+        )
+        preSignedURL = file.preSignedURL
       }
-    }).catch(error => {
+    } catch (e) {
+      console.log("Fail to get file download link: ", e)
+    }
+
+    if (!preSignedURL) {
       console.log("Fail to get file download link")
-    })
+    }
+    window.open(preSignedURL)
+  }
+
+  const hasFileHandle = (fh:FileHandle) => {
+    if (fh && !fh['isPreview']) {
+      setHasFileAccess(true)
+      return true
+    } else {
+      setHasFileAccess(false)
+      return false
+    }
+  }
+
+  const getTableEntityFileHandle = () => {
+    const fileHandleAssociationList: FileHandleAssociation[] = [
+      {
+        associateObjectId: associatedObjectId!,
+        associateObjectType: associatedObjectType!,
+        fileHandleId: fileHandleId!,
+      },
+    ]
+    const batchFileRequest: BatchFileRequest = {
+      includeFileHandles: true,
+      includePreSignedURLs: true,
```

Comment:
Can includePreSignedURLs be set to false here?

Reply:
Updated. Thanks.